OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/FirebaseAnalytics.java

```diff
@@ -0,0 +1,378 @@
+                                @Nullable Map<String, String> values) {
+        final FirebaseEvent event = new FirebaseEvent(screenName);
+        if (!TextUtils.isEmpty(action)) {
+            event.putString(Keys.ACTION, action);
+        }
+        if (!TextUtils.isEmpty(courseId)) {
+            event.putCourseId(courseId);
+        }
+        if (values != null) {
+            event.putMap(values);
+        }
+
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackVideoLoading(String videoId, String courseId, String unitUrl) {
+        final FirebaseEvent event = new FirebaseEvent(Events.LOADED_VIDEO, videoId, Values.VIDEO_LOADED);
+        event.setCourseContext(courseId, unitUrl, Values.VIDEOPLAYER);
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackVideoPlaying(String videoId, Double currentTime,
+                                  String courseId, String unitUrl) {
+        final FirebaseEvent event = new FirebaseEvent(Events.PLAYED_VIDEO, videoId,
+                Values.VIDEO_PLAYED, currentTime);
+        event.setCourseContext(courseId, unitUrl, Values.VIDEOPLAYER);
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackVideoPause(String videoId,
+                                Double currentTime, String courseId, String unitUrl) {
+        final FirebaseEvent event = new FirebaseEvent(Events.PAUSED_VIDEO,
+                videoId, Values.VIDEO_PAUSED, currentTime);
+        event.setCourseContext(courseId, unitUrl, Values.VIDEOPLAYER);
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackVideoStop(String videoId, Double currentTime, String courseId,
+                               String unitUrl) {
+        final FirebaseEvent event = new FirebaseEvent(Events.STOPPED_VIDEO,
+                videoId, Values.VIDEO_STOPPED, currentTime);
+        event.setCourseContext(courseId, unitUrl, Values.VIDEOPLAYER);
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackVideoSeek(String videoId, Double oldTime, Double newTime, String courseId,
+                               String unitUrl, Boolean skipSeek) {
+        final FirebaseEvent event = new FirebaseEvent(Events.SEEK_VIDEO, videoId, Values.VIDEO_SEEKED);
+        event.setCourseContext(courseId, unitUrl, Values.VIDEOPLAYER);
+        //Call the format Double value so that we can have upto 3 decimal places after
+        oldTime = JavaUtil.formatDoubleValue(oldTime, 3);
+        newTime = JavaUtil.formatDoubleValue(newTime, 3);
+        Double skipInterval = newTime - oldTime;
+        skipInterval = JavaUtil.formatDoubleValue(skipInterval, 3);
+        event.putDouble(Keys.OLD_TIME, oldTime);
+        event.putDouble(Keys.NEW_TIME, newTime);
+        if (skipSeek) {
```

Comment:
I don't think this logic should belong in here. I was thinking one level up so that Segment and Firebase are DRY free, but not sure if it makes sense to put this in the AnalyticsRegistery